### PR TITLE
[stdlib]Remove unneeded @inlinable attributes from the array casts

### PR DESCRIPTION
**Pull request by lancep**

### CODE REVIEW

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please benchmark

File: stdlib/public/core/ArrayCast.swift [hunk 1]
```diff
@@ -60,7 +59,7 @@ internal struct _UnwrappingFailed : Error {
 }
 
 extension Optional {
-  @inlinable // FIXME(sil-serialize-all)
+  @usableFromInline
   internal func unwrappedOrError() throws -> Wrapped {
     if let x = self { return x }
     throw _UnwrappingFailed()
```
Comment (jrose-apple) [inline stdlib/public/core/ArrayCast.swift]: This is clever but just writing the loop out long-hand is probably better regardless.

Comment (lancep) [reply stdlib/public/core/ArrayCast.swift]: Agreed!

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please test

**Comment by lancep** [comment]
@swift-ci please test OS X platform

**Comment by lancep** [comment]
@swift-ci please test OS X platform

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please benchmark

**Pull request merged**